Realm/RLMPredicateUtil.mm

```diff
@@ -72,8 +74,12 @@ - (NSExpression *)falseExpression;
         case NSMinusSetExpressionType:
             return [NSExpression expressionForMinusSet:visit(expression.leftExpression) with:visit(expression.rightExpression)];
 
-        case NSSubqueryExpressionType:
-            return [NSExpression expressionForSubquery:visit(expression.operand) usingIteratorVariable:expression.variable predicate:visit(expression.predicate)];
+        case NSSubqueryExpressionType: {
+            NSExpression *collection = expression.collection;
+            // NSExpression.collection is declared as id, but appears to always hold an NSExpression for subqueries.
+            REALM_ASSERT([collection isKindOfClass:[NSExpression class]]);
```

Comment:
I went with `REALM_ASSERT` here because we're only likely to discover this assumption is wrong when a user runs into this, and users don't often use debug versions of the Realm framework.